gravitee-node-notifier/src/main/java/io/gravitee/node/notifier/NotifierServiceImpl.java

```diff
@@ -75,19 +82,73 @@ public void register(
       );
       final String id = buildNotificationId(
         definition.getResourceId(),
+        definition.getResourceType(),
         definition.getType(),
         definition.getAudienceId()
       );
       triggers.put(id, notificationTrigger);
+      preserveNotificationDefinition(definition);
+
       notificationTrigger.start();
     } else {
       logger.debug("Node notifier service disabled");
     }
   }
 
+  private void preserveNotificationDefinition(
+    NotificationDefinition definition
+  ) {
+    final String resourceKey = buildResourceKey(
+      definition.getResourceId(),
+      definition.getResourceType()
+    );
+    List<NotificationDefinition> allDefinitions = definitions.get(resourceKey);
+    if (Objects.isNull(allDefinitions)) {
+      allDefinitions = Collections.synchronizedList(new ArrayList<>());
+      definitions.put(resourceKey, allDefinitions);
+    }
+    allDefinitions.add(definition);
```

Comment:
Maybe I'm wrong but I suspect a possible side effect if 2 `NotificationDefinition` on the same resource are registered almost in the same time (the `allDefinitions` list could be erased and one of the NotificationDefinition can be lost).

Do you think it is possible to rework it using a `computeIfAbsent` instead and rely on the concurrent map to ensure nothing could be lost ?